src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -2395,5 +2395,36 @@ private bool notifyMethodInfoUsage(CORINFO_METHOD_STRUCT_* ftn)
         {
             return true;
         }
+
+        private CORINFO_CLASS_STRUCT_* getStaticFieldCurrentClass(CORINFO_FIELD_STRUCT_* field, byte* pIsSpeculative)
```

Comment:
I'm a bit confused about the expected semantics of this API.

```
// Return field's exact type if its value is known to be never changed and
// is not null (for reference types). Returns nullptr otherwise.
```

Like Jan said - we don't check the initonly part - neither for the RVA nor for non-RVA statics. Should this always return null for `!initonly`?
We also return the type of the field, not the type of the value stored in the field (in the reference type case).